Code/autopkglib/github/__init__.py

```diff
@@ -89,6 +97,69 @@ def setup_token(self):
 
         self.token = token
 
+    def prepare_curl_cmd(self, method, accept, headers, data, temp_content):
+        """Assemble curl command and return it."""
+        curl_cmd = [
+            super(GitHubSession, self).curl_binary(),
+            "--location",
+            "--silent",
+            "--show-error",
+            "--fail",
+            "--dump-header",
+            "-",
+        ]
+
+        curl_cmd.extend(["-X", method])
+        curl_cmd.extend(["--header", "%s: %s" % ("User-Agent", "AutoPkg")])
+        curl_cmd.extend(["--header", "%s: %s" % ("Accept", accept)])
+
+        # Pass the GitHub token as a header
+        if self.token:
+            curl_cmd.extend(
+                ["--header", "%s: %s" % ("Authorization", "token %s" % self.token)]
+            )
+
+        super(GitHubSession, self).add_curl_common_opts(curl_cmd)
+
+        # Additional headers if defined
+        if headers:
+            for header, value in headers.items():
+                curl_cmd.extend(["--header", "%s: %s" % (header, value)])
+
+        # Set the data header if defined
+        if data:
+            data = json.dumps(data)
+            curl_cmd.extend(["-d", data, "--header", "Content-Type: application/json"])
+
+        # Final argument to curl is the URL
+        curl_cmd.extend(["--url", self.env["url"]])
+        curl_cmd.extend(["--output", temp_content])
```

Comment:
We are now saving the data output to the temporary file so headers cant be parsed from curl output via `URLGetter`.